test/Microsoft.IdentityModel.Tokens.Tests/CryptoProviderFactoryTests.cs

```diff
@@ -507,6 +511,194 @@ public static TheoryData<SignatureProviderTheoryData> ReleaseSignatureProvidersT
                     {
                         SigningSignatureProvider = symmetricSignatureProviderCached,
                         TestId = "Release4"
+                    },
+                    new SignatureProviderTheoryData
+                    {
+                       CustomCryptoProvider = new CustomCryptoProvider(new string[] {"RS256"})
+                       {
+                           SignatureProvider = asymmetricSignatureProviderToRelease
+                       },
+                       SigningSignatureProvider = asymmetricSignatureProviderToRelease,
+                       TestId = "CustomCryptoProviderRelease"
+                    },
+                    new SignatureProviderTheoryData
+                    {
+                       ExpectedException = EE.ArgumentNullException(),
+                       CustomCryptoProvider = new CustomCryptoProvider(new string[] {"RS256"})
+                       {
+                           SignatureProvider = asymmetricSignatureProviderToRelease
+                       },
+                       SigningSignatureProvider = null,
+                       TestId = "CustomCryptoProviderRelease - SignatureProvider null"
+                    }
+                };
+
+                return theoryData;
+            }
+        }
+
+        [Theory, MemberData(nameof(ReleaseHashAlgorithmsTheoryData))]
+        public void ReleaseHashAlgorithms(CryptoProviderFactoryTheoryData theoryData)
+        {
+            IdentityModelEventSource.ShowPII = true;
```

Comment:
the constructor to TheoryDataBase has code: IdentityModelEventSource.ShowPII = true;